judge/judgedaemon.main.php

```diff
@@ -850,55 +880,50 @@ function judge(array $row)
-        $message = 'No files with allowed extensions found to pass to compiler. Allowed extensions: ' . implode(', ', $row['language_extensions']);
+        // Note: It may be tempting to assume that this codepath can be never
+        // reached since we prevent these submissions from being submitted both
+        // via command line and the web interface. However, the code path can
+        // be triggered when the filtering is activated between submission and
+        // rejudge.
+        $message = 'No files with allowed extensions found to pass to compiler. Allowed extensions: '
+            . implode(', ', $row['language_extensions']);
         $args = 'compile_success=0' .
-                '&output_compile=' . urlencode(base64_encode($message));
+            '&output_compile=' . urlencode(base64_encode($message));
 
         $url = sprintf('judgehosts/update-judging/%s/%s', urlencode($myhost), urlencode((string)$row['judgingid']));
         request($url, 'PUT', $args);
 
-        // revoke readablity for domjudge-run user to this workdir
+        // Revoke readablity for domjudge-run user to this workdir.
         chmod($workdir, 0700);
         logmsg(LOG_NOTICE, "Judging s$row[submitid]/j$row[judgingid]: compile error");
-        return;
+        return false;
     }
 
     if (count($files)==0) {
         error("No submission files could be downloaded.");
     }
 
-    if (empty($row['compile_script'])) {
-        error("No compile script specified for language " . $row['langid'] . ".");
-    }
-
     list($execrunpath, $error) = fetch_executable(
         $workdirpath,
-        $row['compile_script'],
-        $row['compile_script_md5sum']
+        'compile',
+        $judgeTask['compile_script_id']
     );
     if (isset($error)) {
-        logmsg(LOG_ERR, "fetching executable failed for compile script '" . $row['compile_script'] . "': " . $error);
-        $description = $row['compile_script'] . ': fetch, compile, or deploy of compile script failed.';
-        disable('language', 'langid', $row['langid'], $description, $row['judgingid'], (string)$row['cid']);
-        return;
-    }
-
-    // create chroot environment
-    logmsg(LOG_INFO, "  √ Executing chroot script: '".CHROOT_SCRIPT." start'");
-    system(LIBJUDGEDIR.'/'.CHROOT_SCRIPT.' start', $retval);
-    if ($retval!=0) {
-        error("chroot script exited with exitcode $retval");
+        logmsg(LOG_ERR, "Fetching executable failed for compile script '" . $judgeTask['compile_script_id'] . "': " . $error);
+        $description = $judgeTask['compile_script_id'] . ': fetch, compile, or deploy of compile script failed.';
+        disable('compile_script', 'compile_script_id', $judgeTask['compile_script_id'], $description, $judgeTask['judgetaskid']);
+        return false;
     }
 
     // Compile the program.
     system(LIBJUDGEDIR . "/compile.sh $cpuset_opt '$execrunpath' '$workdir' " .
-           implode(' ', $files), $retval);
+        implode(' ', $files), $retval);
```

Comment:
:thinking: does this expose us to shell injection vulnerability if you choose a clever filename and upload it? I assume that is probably filtered elsewhere, but it occurred to me while reading this we might want to do some shell escaping magic.

Reply:
Filenames are restricted in: https://github.com/DOMjudge/domjudge/blob/master/webapp/src/Service/SubmissionService.php#L30

But I agree that we should also add escaping here (in another PR).

Reply:
yeah, I definitely don't think we need to do that as part of this PR. Opened https://github.com/DOMjudge/domjudge/issues/960 to track this.